upgrading/topics/keycloak/changes-18_0_0.adoc

```diff
@@ -0,0 +1,20 @@
+= Step-up authentication
+
+Step-up authentication is new feature added. There is `acr` client scope, which contains protocol mapper, which is supposed to add `acr`
+claim in the token. So the `acr` claim is not added automatically now as it was before this version, but it is added with the usage
+of this client scope and protocol mapper.
+
+The client scope is added as realm "default" client scope and hence will be added to all newly created clients. But for performance reasons,
+the client scope is not automatically added to all existing clients during migration. So clients won't have `acr` claim by default after
+the migration. Few possibilities to handle this:
+
+- If you don't plan to use step-up authentication feature, but you rely on the `acr` claim in the token, you can disable `step_up_authentication`
+  feature as described in the link:{installguide_link}#profiles[{installguide_name}]. The claim will be added with value `1` in case of normal authentication and `0` in case of SSO authentication.
+- Add `acr` client scope to your clients manually by admin REST API or admin console. This is needed especially if you want to use step-up authentication.
+  If you have bigger amount of clients in the realm and want to use `acr` claim for all of them, you can trigger some SQL similar to this against your DB (don't forget to clear the cache
```

Comment:
It might be good that L14 and L15 be one sentence.

Reply:
This is already one sentence though?

Reply:
Sorry. What I wanted to say is to remove linefeeds among L14-L15. Not write as them one sentence.

Reply:
Ah, I understand now. Thanks for the clarification.

But TBH I don't think that removing linefeeds is good thing neither as it means that particular line can be inadequately long when you edit it in the IDE/text editor, which IMO makes it harder to edit and maintain. 

One point to note here is, that these line feeds are shown just here in the "adoc" file. When the final documentation is built/rendered, the line feeds from "adoc" file are ignored and the final format (EG. HTML) does it's own line feeds on different places to adjust the page. For example see this section in the "adoc" sources https://github.com/keycloak/keycloak-documentation/blame/main/upgrading/topics/keycloak/changes-16_0_0.adoc#L19-L22 and compare line feeds with the final rendered HTML here: https://www.keycloak.org/docs/latest/upgrading/index.html#proxy-environment-variables .

Reply:
@mposolda Yes, I agree.

Reply:
@tnorimat Thanks